ui/app/AppLayouts/Wallet/controls/FollowingAddressesDelegate.qml

```diff
@@ -0,0 +1,284 @@
+            if (address.toLowerCase() === root.address.toLowerCase()) {
+                d.savedAddressesVersion++
+            }
+        }
+        
+        function onSavedAddressDeleted(name, address, errorMsg) {
+            if (address.toLowerCase() === root.address.toLowerCase()) {
+                d.savedAddressesVersion++
+            }
+        }
+    }
+
+    onClicked: {
+        if (root.usage === FollowingAddressesDelegate.Usage.Item) {
+            return
+        }
+        Global.openSavedAddressActivityPopup({
+                                                 name: root.name,
+                                                 address: root.address,
+                                                 ens: root.ensName,
+                                                 colorId: "",
+                                                 avatar: root.avatar,
+                                                 isFollowingAddress: true
+                                              })
+    }
+
+    components: [
+        StatusRoundButton {
+            id: sendButton
+            visible: !!root.name && root.showButtons
+            type: StatusRoundButton.Type.Quinary
+            radius: 8
+            icon.name: "send"
+            enabled: root.networkConnectionStore.sendBuyBridgeEnabled
+            onClicked: root.openSendModal(root.address)
+        },
+        StatusRoundButton {
+            id: starButton
+            visible: !!root.name && (d.isAddressSaved || root.showButtons)
+            type: StatusRoundButton.Type.Quinary
+            radius: 8
+            icon.name: d.isAddressSaved ? "star-icon" : "star-icon-outline"
+            enabled: !d.isAddressSaved
+            onClicked: {
+                var nameToUse = root.ensName || root.address
+                if (root.ensName && root.ensName.includes(".")) {
+                    nameToUse = root.ensName.split(".")[0]
+                }
+                
+                Global.openAddEditSavedAddressesPopup({
+                    addAddress: true,
+                    address: root.address,
+                    name: nameToUse,
+                    ens: root.ensName
+                })
+            }
+        },
+        StatusRoundButton {
+            objectName: "followingAddressView_Delegate_menuButton_" + root.name
+            visible: !!root.name
+            enabled: root.showButtons
+            type: StatusRoundButton.Type.Quinary
+            radius: 8
+            icon.name: "more"
+            onClicked: {
+                menu.openMenu(this, x + width - menu.width - statusListItemComponentsSlot.spacing, y + height + Theme.halfPadding,
+                    {
+                        name: root.name,
+                        address: root.address,
+                        ensName: root.ensName,
+                        tags: root.tags,
+                    }
+                );
+            }
+
+        }
+    ]
+
+    StatusMenu {
```

Comment:
I'd suggest not making the menu part of the delegate, extract it, wrap it in a Component and handle it externally (from the place where it's actually used)